src/scvi/module/_scanvae.py

```diff
@@ -324,15 +323,26 @@ def loss(
         if self.classifier.logits:
             probs = F.softmax(probs, dim=-1)
 
-        reconst_loss += loss_z1_weight + (
-            (loss_z1_unweight).view(self.n_labels, -1).t() * probs
-        ).sum(dim=1)
-
-        kl_divergence = (kl_divergence_z2.view(self.n_labels, -1).t() * probs).sum(dim=1)
-        kl_divergence += kl(
-            Categorical(probs=probs),
-            Categorical(probs=self.y_prior.repeat(probs.size(0), 1)),
-        )
+        if self.n_labels > 1:
+            reconst_loss += loss_z1_weight + (
+                (loss_z1_unweight).view(self.n_labels, -1).t() * probs
+            ).sum(dim=1)
```

Comment:
.sum(dim=-1)

Reply:
Remove this check with self.n_labels>1 again.

Reply:
This comes from the previous solution, which was fine, for the case of n_labels>1 and n_samples==1